### ISSUE TITLE
Restore permissions not applying to folders

### ISSUE THREAD

**Issue by Tralen** [duplicati/duplicati#2512]
I have:
- [x] searched open and closed issues for duplicates

----------------------------------------

### Version info

**Duplicati Version:** 2.0.1.60_canary_2017-05-23
**Operating System:** Arch Linux on kernel 4.9.27-1-lts
**Backend:** Amazon Cloud Drive

### Bug description
When restoring files, across multiple folders, the setting "Restore read/write permissions" applies properly to the files recovered, but not to the folders they are in. All folders in the given filepaths are set to root, not to their original user.

### Steps to reproduce
- Select to "Restore Files".
- Select files across *different* folders, to force the creation of the containing folders.
- Select to "Pick location", so that the original folders are not used.
- Check the option to "Restore read/write permissions"
- Attempt to restore the files.

**Actual result:**
The files will be created with proper permissions. The folders won't, they will be given to root.

**Expected result:**
Everything, including the folders, should have the original permissions.

**Comment by kenkendk** [comment]
It appears I forgot to apply the permission to the directories after the restore. Should be simple to fix.

**Comment by warwickmm** [comment]
I am unable to reproduce this with version 2.0.2.19_canary_2018-02-12.  I created a simple backup of two folders with different owners and different `rwx` permissions.  Both folders contain a file with different owners and different `rwx` permissions.  I was able to restore this to a different location and the owners/permissions of the contents were the same as the source files/folders.  I am using a local backend as opposed to Amazon Cloud Drive, but I don't think this should depend on the backend.

@Tralen, are you still experiencing this issue?
